tpmutil/run_windows.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2018, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tpmutil
+
+import (
+	"fmt"
+	"io"
+	"syscall"
+	"unsafe"
+)
+
+// Tbs.dll Docs:
+// https://docs.microsoft.com/en-us/windows/desktop/TBS/tpm-base-services-portal
+var tbsDll = syscall.NewLazyDLL("Tbs.dll")
+var tbsCreateContext = tbsDll.NewProc("Tbsi_Context_Create")
+var tbsSubmitCommand = tbsDll.NewProc("Tbsip_Submit_Command")
+var tbsContextClose = tbsDll.NewProc("Tbsip_Context_Close")
+
+// tbsContextParams2 Docs:
+// https://docs.microsoft.com/en-us/windows/desktop/api/Tbs/ns-tbs-tdtbs_context_params2
+type tbsContextParams2 struct {
+	version uint32
+	flags   uint32
+}
+
+const (
+	tpm2Version            uint32  = 2
+	bothTPMVersionsFlag    uint32  = 6
+	tbsCommandLocalityZero uintptr = 0
+)
+
+// CommandPriority Parameter Docs
+// https://docs.microsoft.com/en-us/windows/desktop/api/Tbs/nf-tbs-tbsip_submit_command#parameters
+type CommandPriority uint32
+
+// CommandPriority is used to determine which pending command to submit whenever the TPM is free.
+// https://docs.microsoft.com/en-us/windows/desktop/tbs/command-scheduling
+const (
+	LowPriority     CommandPriority = 100          // used for low priority application use.
+	NormalPriority  CommandPriority = 200          // used for normal priority application use.
+	HighPriority    CommandPriority = 300          // used for high priority application use.
+	SystemPriority  CommandPriority = 400          // used for system tasks that access the TPM.
+	DefaultPriority                 = HighPriority // uses HighPriority as default
+)
+
+// Error Codes:
+// https://docs.microsoft.com/en-us/windows/desktop/TBS/tbs-return-codes
+var errMap = map[uintptr]string{
+	0x80284001: "An internal software error occurred.",
```

Comment:
[optional]: The integer constants might be clearer if they were written as constants using iota, since then it would be clear that 0x80284000 is the prefix and the errors follow in order.